File: print-in-order/print-in-order.c
typedef struct {
    // User defined data may be declared here.
    int lock;
} Foo;

Foo* fooCreate() {
    Foo* obj = (Foo*) malloc(sizeof(Foo));
    
    // Initialize user defined data here.
    obj->lock = 1;
    return obj;
}

void first(Foo* obj) {
    
    while (obj->lock != 1) {}
    // printFirst() outputs "first". Do not change or remove this line.
    printFirst();
    obj->lock++;
}

void second(Foo* obj) {
    
    while (obj->lock != 2) {}
    // printSecond() outputs "second". Do not change or remove this line.
    printSecond();
    obj->lock++;
}

void third(Foo* obj) {
    
    while (obj->lock != 3) {}
    // printThird() outputs "third". Do not change or remove this line.
    printThird();
    obj->lock++;
}

void fooFree(Foo* obj) {
    // User defined data may be cleaned up here.
    
}

File: print-in-order/print-in-order.py
class Foo:
    def __init__(self):
        self.lock = 1


    def first(self, printFirst: 'Callable[[], None]') -> None:
        
        # printFirst() outputs "first". Do not change or remove this line.
        while self.lock != 1:
            pass
        printFirst()
        self.lock += 1


    def second(self, printSecond: 'Callable[[], None]') -> None:
        
        # printSecond() outputs "second". Do not change or remove this line.
        while self.lock != 2:
            pass
        printSecond()
        self.lock += 1


    def third(self, printThird: 'Callable[[], None]') -> None:
        
        # printThird() outputs "third". Do not change or remove this line.
        while self.lock != 3:
            pass
        printThird()
        self.lock += 1